### ISSUE TITLE
CA and auth setup improvements

### ISSUE THREAD

**Issue by costinm** [istio-ecosystem/istio-installer#34]
1. Default to 'provisioned cert', use the tool to generate the root and upload it - for consistency 
and to make multi-cluster setup easier. This will eliminate some options and simplify the templates.
2. Extend the tooling to generate a .CRT and .PFX file that can be imported in browsers
3.  Switch grafana/prom/etc access via gateway to use client certs and istio-based RBAC, using the generate CRT/PFX

**Comment by loewenstein** [comment]
@costinm what is 'the tool'? You are referring to the installer, i.e. the `iop` script? Is a feature like this - cert creation, upload, browser cert creation - a good enough reason to switch from hacking bash to writing well tested Go?

**Comment by costinm** [comment]
'the tool' will be istioctl - the iop script and the bash are mostly examples that can be used for users who have existing pipelines and CI/CD systems. Operator will be in go.

It is not clear if the tool will be in a separate repo ( i.e. the operator ) or we need to have it in this repo.
I prefer separate repo, keeping this one focused on the templates. 

In either case - we must support customers who want to integrate istio with their frameworks/environments - we can't require anyone to use operator/iop. As long as anything that the go program is doing is well documented and can be replicated in a small shell script or other tool - we are good.

**Comment by costinm** [comment]
BTW - istio/istioclt can link depenencies from different repos - we can develop all install/operator code in 
a separate repo, with a 'main' that allows testing - and then link it with istioctl. 
Better to keep things modular.

**Comment by loewenstein** [comment]
> I prefer separate repo, keeping this one focused on the templates
Agreed. So we would need that repo to work on this issue, didn't we? And for certs I would expect there is something reusable in the Citadel code. Although that would make a separate repo complicated, maybe a reason to duplicate instead of adding a dependency...
